File: FrogBaek/49_Group_Anagrams.cpp
class Solution
{
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs)
    {
        unordered_map<string, vector<string>> m;
        for (auto& str : strs)
        {
            string t = str;
            sort(t.begin(), t.end());
            m[t].push_back(str);
        }

        vector<vector<string>> ans;
        for (auto& p : m)
            ans.push_back(p.second);

        return ans;        
    }
};

File: alexgoni/49_Group_Anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function (strs) {
  const dict = {};

  for (const str of strs) {
    const sortedStr = [...str].sort().join("");

    if (dict[sortedStr]) {
      dict[sortedStr].push(str);
    } else {
      dict[sortedStr] = [str];
    }
  }

  return Object.values(dict);
};
